backends/generators/sverilog/sverilog_generator.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import sys
+import logging
+from pathlib import Path
+
+# Add parent directory to path for imports
+sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+
+from generator import load_instructions, load_csrs
+
+
+def format_instruction_name(name):
+    """Format instruction name for SystemVerilog (uppercase with underscores)."""
+    # Handle compressed instructions
+    if name.startswith("c."):
+        name = "C_" + name[2:]
+    # Replace dots with underscores and convert to uppercase
+    return name.replace(".", "_").upper()
+
+
+def format_csr_name(name):
+    """Format CSR name for SystemVerilog (uppercase with underscores)."""
+    return "CSR_" + name.replace(".", "_").upper()
+
+
+def match_to_sverilog_bits(match_str, is_compressed=False):
+    """Convert a match string to SystemVerilog bit pattern."""
+    if not match_str:
+        return "32'b" + "?" * 32
+    
+    # For compressed instructions (16-bit), we need to handle them differently
+    # The riscv-opcodes format puts the 16-bit pattern in the lower 16 bits
+    # with the upper 16 bits as wildcards
+    if is_compressed or len(match_str) == 16:
+        # Pad with wildcards on the left for 16-bit instructions
+        match_str = "?" * 16 + match_str
+    elif len(match_str) < 32:
+        # For other cases, pad on the right
+        match_str = match_str + "-" * (32 - len(match_str))
+    
+    # Convert to SystemVerilog format (0, 1, or ?)
+    result = []
+    for bit in match_str:
+        if bit == "0":
+            result.append("0")
+        elif bit == "1":
+            result.append("1")
+        else:  # '-' or any other character
+            result.append("?")
+    
+    return "32'b" + "".join(result)
+
+
+def generate_sverilog(instructions, csrs, output_file):
+    """Generate SystemVerilog package file."""
+    with open(output_file, "w") as f:
+        # Write header
+        f.write("\n/* Automatically generated by parse_opcodes */\n")
```

Comment:
Maybe this is more clear?
```suggestion
        f.write("\n/* Automatically generated by UDB */\n")
```